packages/camera/camera_android/android/src/main/java/io/flutter/plugins/camera/Camera.java

```diff
@@ -1293,10 +1293,12 @@ void setImageStreamImageAvailableListener(final EventChannel.EventSink imageStre
   }
 
   void closeCaptureSession() {
-    if (captureSession != null) {
+    // Keep a local copy to avoid race conditions between threads.
+    final CameraCaptureSession captureSessionToClose = captureSession;
+    if (captureSessionToClose != null) {
       Log.i(TAG, "closeCaptureSession");
 
-      captureSession.close();
+      captureSessionToClose.close();
```

Comment:
How does this fix the race condition? If I'm understanding correctly, the premise is that we are calling this method simultaneously from two different threads. If that's true, then this version will call `close` twice on the same session, which will almost certainly just replace a null pointer exception with state error exception.

If this class is trying to interact with the same objects from multiple threads at the same time, that needs to actually be fixed in some robust way (most likely either locking or cross-thread redispatch). Changing individual variable access is almost never the right way to handle threading issues, and that seems to apply here as well.